qnap/shared/teliod.cgi

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+echo "Content-Type: application/json"
+echo ""
+
+# Varibles set by the HTTP web server
+#
+# $REQUEST_METHOD
+# $QUERY_STRING
+# $CONTENT_LENGTH
+
+POST_DATA=""
+
+if [[ "$REQUEST_METHOD" == "POST" || "$REQUEST_METHOD" == "PATCH" ]]; then
+    read -n $CONTENT_LENGTH POST_DATA
+fi
+
+QPKG_DIR="/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet"
+APP_TMP_DIR="/tmp/nordsecuritymeshnet"
+APP_CMD="$QPKG_DIR/teliod"
+CONFIG_FILE="$QPKG_DIR/teliod.cfg"
+LOG_FILE="$APP_TMP_DIR/teliod.log"
+PID_FILE="$APP_TMP_DIR/teliod.pid"
+
+is_application_running() {
+    if [ -f ${PID_FILE} ]; then
+        PID=$(cat ${PID_FILE})
+        if [ -d /proc/${PID}/ ]; then
+            echo 1
+        fi
+    fi
+}
+
+send_response() {
+    local code=$1
+    local message=$2
+    echo "{\"code\": $code, \"message\": \"$message\"}"
+    exit $code
+}
+
+start_application() {
+    if [ is_application_running -eq 1 ]; then
+            send_response 400 "Application is already running."
+    fi
+    $APP_CMD daemon $CONFIG_FILE &
+    if [[ $? -eq 0 ]]; then
+        echo $! > ${PID_FILE}
+        send_response 200 "Application started successfully."
+    else
+        echo $! > ${PID_FILE}
+        send_response 500 "Failed to start the application."
+    fi
+}
+
+stop_application() {
+    if [ -f ${TELIOD_PID_FILE} ]; then
+        PID=$(cat ${TELIOD_PID_FILE})
+        kill -9 ${PID} || true
```

Comment:
why would kill fail here? maybe we need to do something here if process is not successfully killed, instead of leaving it hanging? asking this, since I'm not sure if removing PID file will actually remove process

Reply:
removing PID file wouldn't kill the process by itself. `kill` command might fail if the process is not running.

Reply:
But in this case, if process is not actually running, shouldn't we send 400 as response like in the else statement. Also abit weird that PID file exists, but process doesn't. If this behaviour expected, add some comment saying that PID file can exist, but process might be dead. Idk how to express myself on this, but it just raises questions :smile:

Reply:
> Also abit weird that PID file exists, but process doesn't.

When `teliod` crashes the PID file is not cleaned, that is handled by the start/stop script, when the user enable/disable the app on the QTS (QNAP OS).

Reply:
ok, this part seems to be OK. Can the file cleanup itself be handled from `teliod` itself, whenever crash happens? is it some sort of bug or expected crash behavior which we cannot do anything about?

Reply:
the file only stores the PID, doesn't necessarily mean that the process is running. I doesn't have to.. for that we can check with /proc/<PID> 

not sure if we can/should do it on `teliod`